### PR TITLE
Resolve 215 - Add constant for realm config keys

### PR DESCRIPTION
**Pull request by Donatien26**
Signed-off-by: Donatien Eneman <donatien.eneman@insee.fr>

### CODE REVIEW

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/configuration/ConstantConfig.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  public static final String MUST_CHANGE_PASSWORD = "mustChangePassword";
+  public static final String PASSWORD_CHANGE_REQUEST = "pcr";
+  public static final String PASSWORD = "password";
+
+  public static final String TYPE = "type";
+
+  public static final String USER_SOURCE = "user_source";
+  public static final String APPLI_SOURCE = "application_source";
+  public static final String ORGANIZATION_SOURCE = "organization_source";
```
Comment (CChemin) [inline sugoi-api-core/src/main/java/fr/insee/sugoi/core/configuration/ConstantConfig.java:65]: why not using an enum ?

File: sugoi-api-core/src/main/java/fr/insee/sugoi/core/configuration/ConstantConfig.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package fr.insee.sugoi.core.configuration;
+
+public class ConstantConfig {
```
Comment (CChemin) [inline sugoi-api-core/src/main/java/fr/insee/sugoi/core/configuration/ConstantConfig.java:16]: I'd rather name it something more precise, maybe RealmConfigKey ?

File: sugoi-api-file-store-provider/src/main/java/fr/insee/sugoi/store/file/configuration/ConstantConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package fr.insee.sugoi.store.file.configuration;
+
+public class ConstantConfiguration {
+  public static final String USER_SOURCE = "user_source";
+  public static final String APPLI_SOURCE = "application_source";
+  public static final String ORGANIZATION_SOURCE = "organization_source";
```
Comment (micedre) [inline sugoi-api-file-store-provider/src/main/java/fr/insee/sugoi/store/file/configuration/ConstantConfiguration.java:19]: Why not use the ones defined in  `sugoi-api-core/src/main/java/fr/insee/sugoi/core/configuration/ConstantConfig.java` ?

File: sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/config/ConstantConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  public static final String PASSWORD = "password";
+  public static final String POOL_SIZE = "pool_size";
+  public static final String USER_SOURCE = "user_source";
+  public static final String APP_SOURCE = "app_source";
+  public static final String ORGANIZATION_SOURCE = "organization_source";
+  public static final String ADDRESS_SOURCE = "address_source";
+  public static final String GROUP_SOURCE_PATTERN = "group_source_pattern";
+  public static final String GROUP_FILTER_PATTERN = "group_filter_pattern";
+  public static final String REALM_NAME = "realm_name";
+  public static final String TYPE = "TYPE";
```
Comment (micedre) [inline sugoi-api-ldap-utils/src/main/java/fr/insee/sugoi/ldap/utils/config/ConstantConfiguration.java:31]: Same comment as the FileProvider,some constants should be reused when possible. Also the classNameseems too generic, maybe `LdapConstants`

**Review by micedre — CHANGES_REQUESTED** [review]

**Comment by Donatien26** [comment]
So we have A GlobalKeysConfiguration class where we can find commun key between all the config provider (jms, ldap, filter). We also have one class of keys configuration by provider (ldap, file,...) which extends the firstOne.

Moreover there is a GenericKeysConfig for keys not used by a provider and a EventKeysCOnfiguration for the keys use by the sugoi event part

**Review by micedre — APPROVED** [review]

**Pull request merged**